atom/browser/api/lib/web-contents.js

```diff
@@ -106,14 +108,30 @@ let wrapWebContents = function(webContents) {
     };
   }
 
+  const asyncWebFrameMethods = function(requestId, method, ...args) {
+    this.send('ELECTRON_INTERNAL_RENDERER_ASYNC_WEB_FRAME_METHOD', requestId, method, args);
+    ipcMain.once('ELECTRON_INTERNAL_RENDERER_ASYNC_WEB_FRAME_RESPONSE_' + requestId, function(event, result) {
```

Comment:
If we are adding `requestId` to event name, then there is no need to store callback in `responseCallback` anymore:

```js
const asyncWebFrameMethods = function(requestId, method, callback, ...args) {
  ...
  ipcMain.once(`ELECTRON_BLABLA_${requestId}`, function(event, result) {
    callback(result);
  });
};
```